In [1]:
import pandas as pd

In [2]:
file = pd.read_csv('/Users/fiannualamorgan/Documents/GitHub/Historical_Fires_Near_Me/Sections/Bushfire_News_Articles_19th_Century/Transformed_Data/Geocoded_LOCATION_Mappify_API/Output/2543_completed.csv')

The goal is to verify the accuracy of the locations generated through the Mappify API.

In [3]:
file.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9534 entries, 0 to 9533
Data columns (total 19 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   0                          9534 non-null   object 
 1   1                          9534 non-null   object 
 2   2                          9534 non-null   object 
 3   mappify_buildingName       1078 non-null   object 
 4   mappify_numberFirst        6379 non-null   float64
 5   mappify_numberFirstPrefix  0 non-null      float64
 6   mappify_numberFirstSuffix  82 non-null     object 
 7   mappify_numberLast         2 non-null      float64
 8   mappify_numberLastPrefix   0 non-null      float64
 9   mappify_numberLastSuffix   0 non-null      float64
 10  mappify_streetName         9534 non-null   object 
 11  mappify_streetType         9492 non-null   object 
 12  mappify_streetSuffixCode   43 non-null     object 
 13  mappify_suburb             9534 non-null   objec

In [4]:
file.head()

0         1                                    2  \
0   Auckland  LOCATION     18801230-10-8991754 (1 of 1).txt   
1    Grafton  LOCATION     18801230-10-8991754 (1 of 1).txt   
2  Newcastle  LOCATION     18801230-10-8991754 (1 of 1).txt   
3      Cobar  LOCATION     18801230-10-8991754 (1 of 1).txt   
4     Albury  LOCATION  18840314-436-254369866 (1 of 1).txt   

  mappify_buildingName  mappify_numberFirst  mappify_numberFirstPrefix  \
0          COLLIE BLUE                  NaN                        NaN   
1                  NaN                 18.0                        NaN   
2                  NaN                  NaN                        NaN   
3                  NaN                 32.0                        NaN   
4                  NaN                  NaN                        NaN   

  mappify_numberFirstSuffix  mappify_numberLast  mappify_numberLastPrefix  \
0                       NaN                 NaN                       NaN   
1                       NaN                 NaN                       NaN   
2                       NaN                 NaN                       NaN   
3                       NaN                 NaN                       NaN   
4                       NaN                 NaN                       NaN   

   mappify_numberLastSuffix  mappify_streetName mappify_streetType  \
0                       NaN            BUCKLAND               ROAD   
1                       NaN     GRANTON HEIGHTS               ROAD   
2                       NaN  REDCASTLE CEMETERY               ROAD   
3                       NaN               COBAR             STREET   
4                       NaN              AUBURN               ROAD   

  mappify_streetSuffixCode mappify_suburb mappify_state  mappify_postCode  \
0                      NaN       BUCKLAND           QLD              4722   
1                      NaN        GRANTON           TAS              7030   
2                      NaN      REDCASTLE           VIC              3523   
3                      NaN          COBAR           NSW              2835   
4                      NaN         AUBURN           QLD              4413   

   mappify_location_lat  mappify_location_lon  \
0            -24.517989            147.749324   
1            -42.765878            147.219684   
2            -36.792336            144.785825   
3            -31.527505            145.856320   
4            -26.130303            150.651719   

                           mappify_streetAddress  
0  COLLIE BLUE, BUCKLAND ROAD, BUCKLAND QLD 4722  
1      18 GRANTON HEIGHTS ROAD, GRANTON TAS 7030  
2    REDCASTLE CEMETERY ROAD, REDCASTLE VIC 3523  
3                32 COBAR STREET, COBAR NSW 2835  
4                   AUBURN ROAD, AUBURN QLD 4413

In [5]:
file['0'].value_counts()


Melbourne       767
Adelaide        280
England         254
Ballarat        211
London          197
               ... 
Balla             1
Berrlgan          1
Grose Valley      1
Marongo           1
Rockley           1
Name: 0, Length: 2216, dtype: int64

I can see that my spreadsheet has generated a lot of mentions to England and to London that have been coded, so these need to be removed. England is being put in 'Pentland' QLD. This is a Levenstein distance of 2 so I need to reduce this to 1. 

Return total number of placenames that identically match the Mappify returned result. Only 457 out of 9534 are identical results. 

In [6]:
file.loc[file['0'] == file['mappify_suburb']]

0         1                                     2  \
13       BOWRAL  LOCATION    18960114-35-14033182 (3 of 14).txt   
16     BERRIGAN  LOCATION    18960114-35-14033182 (3 of 14).txt   
23       COROWA  LOCATION     18981217-52-71321501 (2 of 6).txt   
71        MOORA  LOCATION    18961221-346-88994627 (1 of 1).txt   
83       ALBANY  LOCATION      18910107-30-3139737 (2 of 2).txt   
...         ...       ...                                   ...   
9296  MELBOURNE  LOCATION   18980208-966-199925646 (2 of 4).txt   
9426  MELBOURNE  LOCATION  18980209-1459-255622029 (1 of 4).txt   
9469    BENDIGO  LOCATION      18960111-13-8887399 (1 of 8).txt   
9483  WELSHPOOL  LOCATION  18930218-319-198032478 (7 of 10).txt   
9510     MULGOA  LOCATION   18950914-52-71222902 (19 of 21).txt   

     mappify_buildingName  mappify_numberFirst  mappify_numberFirstPrefix  \
13                    NaN                148.0                        NaN   
16                  KIETA               1583.0                        NaN   
23                    NaN                  6.0                        NaN   
71                    NaN                  NaN                        NaN   
83                    NaN                  3.0                        NaN   
...                   ...                  ...                        ...   
9296                  NaN                  6.0                        NaN   
9426                  NaN                  6.0                        NaN   
9469  BENDIGO MARKETPLACE                  1.0                        NaN   
9483                  NaN                  NaN                        NaN   
9510                  NaN                408.0                        NaN   

     mappify_numberFirstSuffix  mappify_numberLast  mappify_numberLastPrefix  \
13                         NaN                 NaN                       NaN   
16                         NaN                 NaN                       NaN   
23                         NaN                 NaN                       NaN   
71                         NaN                 NaN                       NaN   
83                         NaN                 NaN                       NaN   
...                        ...                 ...                       ...   
9296                       NaN                 NaN                       NaN   
9426                       NaN                 NaN                       NaN   
9469                       NaN                 NaN                       NaN   
9483                       NaN                 NaN                       NaN   
9510                       NaN                 NaN                       NaN   

      mappify_numberLastSuffix   mappify_streetName mappify_streetType  \
13                         NaN               BOWRAL             STREET   
16                         NaN             BERRIGAN               ROAD   
23                         NaN               CORONA               ROAD   
71                         NaN                MOORE             STREET   
83                         NaN               ALBANY            HIGHWAY   
...                        ...                  ...                ...   
9296                       NaN            MELBOURNE              PLACE   
9426                       NaN            MELBOURNE              PLACE   
9469                       NaN  BENDIGO MARKETPLACE                NaN   
9483                       NaN            WELSHPOOL               ROAD   
9510                       NaN               MULGOA               ROAD   

     mappify_streetSuffixCode mappify_suburb mappify_state  mappify_postCode  \
13                        NaN         BOWRAL           NSW              2576   
16                        NaN       BERRIGAN           NSW              2712   
23                        NaN         COROWA           NSW              2646   
71                        NaN          MOORA            WA              6510   
83                        NaN     

In [7]:
file['0'] = file['0'].str.upper()
file.head()

0         1                                    2  \
0   AUCKLAND  LOCATION     18801230-10-8991754 (1 of 1).txt   
1    GRAFTON  LOCATION     18801230-10-8991754 (1 of 1).txt   
2  NEWCASTLE  LOCATION     18801230-10-8991754 (1 of 1).txt   
3      COBAR  LOCATION     18801230-10-8991754 (1 of 1).txt   
4     ALBURY  LOCATION  18840314-436-254369866 (1 of 1).txt   

  mappify_buildingName  mappify_numberFirst  mappify_numberFirstPrefix  \
0          COLLIE BLUE                  NaN                        NaN   
1                  NaN                 18.0                        NaN   
2                  NaN                  NaN                        NaN   
3                  NaN                 32.0                        NaN   
4                  NaN                  NaN                        NaN   

  mappify_numberFirstSuffix  mappify_numberLast  mappify_numberLastPrefix  \
0                       NaN                 NaN                       NaN   
1                       NaN                 NaN                       NaN   
2                       NaN                 NaN                       NaN   
3                       NaN                 NaN                       NaN   
4                       NaN                 NaN                       NaN   

   mappify_numberLastSuffix  mappify_streetName mappify_streetType  \
0                       NaN            BUCKLAND               ROAD   
1                       NaN     GRANTON HEIGHTS               ROAD   
2                       NaN  REDCASTLE CEMETERY               ROAD   
3                       NaN               COBAR             STREET   
4                       NaN              AUBURN               ROAD   

  mappify_streetSuffixCode mappify_suburb mappify_state  mappify_postCode  \
0                      NaN       BUCKLAND           QLD              4722   
1                      NaN        GRANTON           TAS              7030   
2                      NaN      REDCASTLE           VIC              3523   
3                      NaN          COBAR           NSW              2835   
4                      NaN         AUBURN           QLD              4413   

   mappify_location_lat  mappify_location_lon  \
0            -24.517989            147.749324   
1            -42.765878            147.219684   
2            -36.792336            144.785825   
3            -31.527505            145.856320   
4            -26.130303            150.651719   

                           mappify_streetAddress  
0  COLLIE BLUE, BUCKLAND ROAD, BUCKLAND QLD 4722  
1      18 GRANTON HEIGHTS ROAD, GRANTON TAS 7030  
2    REDCASTLE CEMETERY ROAD, REDCASTLE VIC 3523  
3                32 COBAR STREET, COBAR NSW 2835  
4                   AUBURN ROAD, AUBURN QLD 4413

In [8]:
from fuzzywuzzy import fuzz

/Users/fiannualamorgan/anaconda3/lib/python3.8/site-packages/fuzzywuzzy/fuzz.py:11: UserWarning: Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning
  warnings.warn('Using slow pure-python SequenceMatcher. Install python-Levenshtein to remove this warning')


In [9]:
file['partial ratio'] = file.apply(lambda x: fuzz.partial_ratio(x['0'], x['mappify_suburb']), axis=1)

In [10]:
file.head()


0         1                                    2  \
0   AUCKLAND  LOCATION     18801230-10-8991754 (1 of 1).txt   
1    GRAFTON  LOCATION     18801230-10-8991754 (1 of 1).txt   
2  NEWCASTLE  LOCATION     18801230-10-8991754 (1 of 1).txt   
3      COBAR  LOCATION     18801230-10-8991754 (1 of 1).txt   
4     ALBURY  LOCATION  18840314-436-254369866 (1 of 1).txt   

  mappify_buildingName  mappify_numberFirst  mappify_numberFirstPrefix  \
0          COLLIE BLUE                  NaN                        NaN   
1                  NaN                 18.0                        NaN   
2                  NaN                  NaN                        NaN   
3                  NaN                 32.0                        NaN   
4                  NaN                  NaN                        NaN   

  mappify_numberFirstSuffix  mappify_numberLast  mappify_numberLastPrefix  \
0                       NaN                 NaN                       NaN   
1                       NaN                 NaN                       NaN   
2                       NaN                 NaN                       NaN   
3                       NaN                 NaN                       NaN   
4                       NaN                 NaN                       NaN   

   mappify_numberLastSuffix  mappify_streetName mappify_streetType  \
0                       NaN            BUCKLAND               ROAD   
1                       NaN     GRANTON HEIGHTS               ROAD   
2                       NaN  REDCASTLE CEMETERY               ROAD   
3                       NaN               COBAR             STREET   
4                       NaN              AUBURN               ROAD   

  mappify_streetSuffixCode mappify_suburb mappify_state  mappify_postCode  \
0                      NaN       BUCKLAND           QLD              4722   
1                      NaN        GRANTON           TAS              7030   
2                      NaN      REDCASTLE           VIC              3523   
3                      NaN          COBAR           NSW              2835   
4                      NaN         AUBURN           QLD              4413   

   mappify_location_lat  mappify_location_lon  \
0            -24.517989            147.749324   
1            -42.765878            147.219684   
2            -36.792336            144.785825   
3            -31.527505            145.856320   
4            -26.130303            150.651719   

                           mappify_streetAddress  partial ratio  
0  COLLIE BLUE, BUCKLAND ROAD, BUCKLAND QLD 4722             88  
1      18 GRANTON HEIGHTS ROAD, GRANTON TAS 7030             86  
2    REDCASTLE CEMETERY ROAD, REDCASTLE VIC 3523             78  
3                32 COBAR STREET, COBAR NSW 2835            100  
4                   AUBURN ROAD, AUBURN QLD 4413             67

Evaluate how many placenames are exact matches and how many have a high threshold. Out of 9534 over half, that is, 5532 are exact matches. Placenames that have scored in the 90s are acceptable, however, anything lower is no good. I also still have not eliminated England or London.  90 is the minimun acceptable threshold for inclusion.

In [11]:
file['partial ratio'].value_counts()

100    5532
67      837
86      801
80      513
83      378
71      292
75      280
88      232
78      217
89      178
90       85
82       60
91       55
92       37
93       16
60        8
85        4
77        2
73        2
87        2
43        1
62        1
94        1
Name: partial ratio, dtype: int64

Check the rows that mention England

In [12]:
file[file['0'] == 'ENGLAND'] 

0         1                                     2  \
47    ENGLAND  LOCATION   18660308-558-147267336 (1 of 2).txt   
56    ENGLAND  LOCATION     18741218-35-13336814 (1 of 1).txt   
191   ENGLAND  LOCATION     18650414-35-13103165 (1 of 2).txt   
228   ENGLAND  LOCATION   18820125-185-200665460 (1 of 1).txt   
246   ENGLAND  LOCATION   18960118-698-162826835 (1 of 1).txt   
...       ...       ...                                   ...   
9257  ENGLAND  LOCATION    18910207-290-91536210 (1 of 1).txt   
9288  ENGLAND  LOCATION   18980114-826-166436677 (1 of 1).txt   
9291  ENGLAND  LOCATION      18600215-13-5677208 (1 of 1).txt   
9437  ENGLAND  LOCATION      18650405-13-5735474 (2 of 2).txt   
9493  ENGLAND  LOCATION  18951224-1093-209054122 (1 of 1).txt   

     mappify_buildingName  mappify_numberFirst  mappify_numberFirstPrefix  \
47          KYONG STATION                  NaN                        NaN   
56          KYONG STATION                  NaN                        NaN   
191         KYONG STATION                  NaN                        NaN   
228         KYONG STATION                  NaN                        NaN   
246         KYONG STATION                  NaN                        NaN   
...                   ...                  ...                        ...   
9257        KYONG STATION                  NaN                        NaN   
9288        KYONG STATION                  NaN                        NaN   
9291        KYONG STATION                  NaN                        NaN   
9437        KYONG STATION                  NaN                        NaN   
9493        KYONG STATION                  NaN                        NaN   

     mappify_numberFirstSuffix  mappify_numberLast  mappify_numberLastPrefix  \
47                         NaN                 NaN                       NaN   
56                         NaN                 NaN                       NaN   
191                        NaN                 NaN                       NaN   
228                        NaN                 NaN                       NaN   
246                        NaN                 NaN                       NaN   
...                        ...                 ...                       ...   
9257                       NaN                 NaN                       NaN   
9288                       NaN                 NaN                       NaN   
9291                       NaN                 NaN                       NaN   
9437                       NaN                 NaN                       NaN   
9493                       NaN                 NaN                       NaN   

      mappify_numberLastSuffix mappify_streetName mappify_streetType  \
47                         NaN    PENTLAND ARAMAC               ROAD   
56                         NaN    PENTLAND ARAMAC               ROAD   
191                        NaN    PENTLAND ARAMAC               ROAD   
228                        NaN    PENTLAND ARAMAC               ROAD   
246                        NaN    PENTLAND ARAMAC               ROAD   
...                        ...                ...                ...   
9257                       NaN    PENTLAND ARAMAC               ROAD   
9288                       NaN    PENTLAND ARAMAC               ROAD   
9291                       NaN    PENTLAND ARAMAC               ROAD   
9437                       NaN    PENTLAND ARAMAC               ROAD   
9493                       NaN    PENTLAND ARAMAC               ROAD   

     mappify_streetSuffixCode mappify_suburb mappify_state  mappify_postCode  \
47                        NaN       PENTLAND           QLD              4816   
56                        NaN       PENTLAND           QLD              4816   
191                       NaN       PENTLAND           QLD              4816   
228                       NaN       PENTLAND           QLD              4816   
246                       NaN       PENTLAND           QLD              4816   


In [13]:
# Use loc to remove all mentions of England from dataframe
file = file.loc[file['0'] != 'ENGLAND']
# Use loc to remove all mentions of London from dataframe
file = file.loc[file['0'] != 'LONDON']
# Test to see if they have been removed, if empty should return trye 
file[file['0'] == 'ENGLAND'].empty

True

In [14]:
#Check value counts again
file['0'].value_counts()

MELBOURNE    831
ADELAIDE     348
BALLARAT     211
SANDHURST    197
COLAC        128
            ... 
ST KITT        1
TEO            1
BALA           1
CANMORE        1
WHITTON        1
Name: 0, Length: 2056, dtype: int64

In [15]:
#check accuracy rate again
file['partial ratio'].value_counts()

100    5532
67      638
86      545
80      513
83      378
71      292
75      280
88      232
78      217
89      178
90       85
82       60
91       55
92       37
93       16
60        8
85        4
77        2
73        2
87        2
43        1
62        1
94        1
Name: partial ratio, dtype: int64

In [16]:
#create new column that contains valid entries
file['partial ratio'].loc()
#file['Valid Places'] = file.apply(lambda x: file.loc(['partial ratio'])


In [17]:
file.axes

[Int64Index([   0,    1,    2,    3,    4,    5,    6,    7,    8,    9,
             ...
             9524, 9525, 9526, 9527, 9528, 9529, 9530, 9531, 9532, 9533],
            dtype='int64', length=9079),
 Index(['0', '1', '2', 'mappify_buildingName', 'mappify_numberFirst',
        'mappify_numberFirstPrefix', 'mappify_numberFirstSuffix',
        'mappify_numberLast', 'mappify_numberLastPrefix',
        'mappify_numberLastSuffix', 'mappify_streetName', 'mappify_streetType',
        'mappify_streetSuffixCode', 'mappify_suburb', 'mappify_state',
        'mappify_postCode', 'mappify_location_lat', 'mappify_location_lon',
        'mappify_streetAddress', 'partial ratio'],
       dtype='object')]

Change this to generate output for placenames with less than 90% accuracy 

In [18]:
# remove all entries that do not reach the accuracy threshold of 90 and return those less than 90 into seperate csv
# file = file.loc[file['partial ratio'] >= 90]
file = file.loc[file['partial ratio'] < 90]

In [19]:
file['partial ratio'].value_counts()

67    638
86    545
80    513
83    378
71    292
75    280
88    232
78    217
89    178
82     60
60      8
85      4
87      2
73      2
77      2
43      1
62      1
Name: partial ratio, dtype: int64

In [20]:
#remove extraneous columns
file = file.drop(['mappify_buildingName', 'mappify_numberFirst',
        'mappify_numberFirstPrefix', 'mappify_numberFirstSuffix',
        'mappify_numberLast', 'mappify_numberLastPrefix',
        'mappify_numberLastSuffix', 'mappify_streetName','mappify_streetType',
        'mappify_streetSuffixCode','mappify_postCode','mappify_streetAddress'], axis = 1)

In [21]:
#write to csv 
# file.to_csv('/Users/fiannualamorgan/Documents/GitHub/Historical_Fires_Near_Me/Sections/Bushfire_News_Articles_19th_Century/Transformed_Data/Geocoded_LOCATION_Mappify_API/Output/returned_mappify_less_90_acc.csv')

Now I will reload the cleaned dataset and attach all meta-data by matching on article id

In [22]:
mappify_results = pd.read_csv('/Users/fiannualamorgan/Documents/GitHub/Historical_Fires_Near_Me/Sections/Bushfire_News_Articles_19th_Century/Transformed_Data/Geocoded_LOCATION_Mappify_API/Transformed/Mappify_Results_Fuzzy_Wuzzy_Validation/cleaned_mappify_perfect_results.csv')

In [23]:
#Check all column names
mappify_results.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5726 entries, 0 to 5725
Data columns (total 9 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Unnamed: 0            5726 non-null   int64  
 1   0                     5726 non-null   object 
 2   1                     5726 non-null   object 
 3   2                     5726 non-null   object 
 4   mappify_suburb        5726 non-null   object 
 5   mappify_state         5726 non-null   object 
 6   mappify_location_lat  5726 non-null   float64
 7   mappify_location_lon  5726 non-null   float64
 8   partial ratio         5726 non-null   int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 402.7+ KB


In [24]:
#print first five rows
mappify_results.head()

Unnamed: 0          0         1                                    2  \
0           3      COBAR  LOCATION     18801230-10-8991754 (1 of 1).txt   
1           5   BENAMBRA  LOCATION  18840314-436-254369866 (1 of 1).txt   
2           7   BRIGHTON  LOCATION     18630202-13-6483150 (1 of 1).txt   
3           8  MELBOURNE  LOCATION     18630202-13-6483150 (1 of 1).txt   
4           9     BYWONG  LOCATION     18950225-30-4550862 (2 of 2).txt   

  mappify_suburb mappify_state  mappify_location_lat  mappify_location_lon  \
0          COBAR           NSW            -31.527505            145.856320   
1       BENAMBRA           VIC            -36.972401            147.702019   
2       BRIGHTON            SA            -35.016432            138.523661   
3      MELBOURNE           VIC            -37.813513            144.969082   
4         BYWONG           NSW            -35.169799            149.332208   

   partial ratio  
0            100  
1            100  
2            100  
3            100  
4            100

In [44]:
#need to create a new column from column 2 that returns the unique id so break on comma
article_ids = []
for i,row in mappify_results.iterrows():
    unique_id = row['2'].split(' ')[0].split('-')[-1]
    article_ids.append(unique_id)


#article_id = mappify_results['2'].str.split('(')[0]

In [47]:
#merge these two dataframes and drop column 1 and 0
mappify_results['article_id'] = article_ids
merged_df = mappify_results

In [50]:
merged_df.head()

Unnamed: 0          0         1                                    2  \
0           3      COBAR  LOCATION     18801230-10-8991754 (1 of 1).txt   
1           5   BENAMBRA  LOCATION  18840314-436-254369866 (1 of 1).txt   
2           7   BRIGHTON  LOCATION     18630202-13-6483150 (1 of 1).txt   
3           8  MELBOURNE  LOCATION     18630202-13-6483150 (1 of 1).txt   
4           9     BYWONG  LOCATION     18950225-30-4550862 (2 of 2).txt   

  mappify_suburb mappify_state  mappify_location_lat  mappify_location_lon  \
0          COBAR           NSW            -31.527505            145.856320   
1       BENAMBRA           VIC            -36.972401            147.702019   
2       BRIGHTON            SA            -35.016432            138.523661   
3      MELBOURNE           VIC            -37.813513            144.969082   
4         BYWONG           NSW            -35.169799            149.332208   

   partial ratio article_id  
0            100    8991754  
1            100  254369866  
2            100    6483150  
3            100    6483150  
4            100    4550862

In [53]:
merged_df.columns= ['old_index', 'place_name', 'entity_type', 'filename', 'mappify_suburb', 'state','mappify_lat', 'mappify_long', 'partial_ratio','article_id']
merged_df.head()

old_index place_name entity_type                             filename  \
0          3      COBAR    LOCATION     18801230-10-8991754 (1 of 1).txt   
1          5   BENAMBRA    LOCATION  18840314-436-254369866 (1 of 1).txt   
2          7   BRIGHTON    LOCATION     18630202-13-6483150 (1 of 1).txt   
3          8  MELBOURNE    LOCATION     18630202-13-6483150 (1 of 1).txt   
4          9     BYWONG    LOCATION     18950225-30-4550862 (2 of 2).txt   

  mappify_suburb state  mappify_lat  mappify_long  partial_ratio article_id  
0          COBAR   NSW   -31.527505    145.856320            100    8991754  
1       BENAMBRA   VIC   -36.972401    147.702019            100  254369866  
2       BRIGHTON    SA   -35.016432    138.523661            100    6483150  
3      MELBOURNE   VIC   -37.813513    144.969082            100    6483150  
4         BYWONG   NSW   -35.169799    149.332208            100    4550862

In [54]:
# delete the columns ['remaining_trove_id','old_index', 'mystery', 'second_placename']
merged_df = merged_df.drop(['old_index'], axis=1)
merged_df.head()

place_name entity_type                             filename mappify_suburb  \
0      COBAR    LOCATION     18801230-10-8991754 (1 of 1).txt          COBAR   
1   BENAMBRA    LOCATION  18840314-436-254369866 (1 of 1).txt       BENAMBRA   
2   BRIGHTON    LOCATION     18630202-13-6483150 (1 of 1).txt       BRIGHTON   
3  MELBOURNE    LOCATION     18630202-13-6483150 (1 of 1).txt      MELBOURNE   
4     BYWONG    LOCATION     18950225-30-4550862 (2 of 2).txt         BYWONG   

  state  mappify_lat  mappify_long  partial_ratio article_id  
0   NSW   -31.527505    145.856320            100    8991754  
1   VIC   -36.972401    147.702019            100  254369866  
2    SA   -35.016432    138.523661            100    6483150  
3   VIC   -37.813513    144.969082            100    6483150  
4   NSW   -35.169799    149.332208            100    4550862

In [55]:
merged_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5726 entries, 0 to 5725
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   place_name      5726 non-null   object 
 1   entity_type     5726 non-null   object 
 2   filename        5726 non-null   object 
 3   mappify_suburb  5726 non-null   object 
 4   state           5726 non-null   object 
 5   mappify_lat     5726 non-null   float64
 6   mappify_long    5726 non-null   float64
 7   partial_ratio   5726 non-null   int64  
 8   article_id      5726 non-null   object 
dtypes: float64(2), int64(1), object(6)
memory usage: 402.7+ KB


In [56]:
#try and reassign article_id as an integer
merged_df = merged_df.astype({'article_id': int})

In [57]:
#check this has worked
merged_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5726 entries, 0 to 5725
Data columns (total 9 columns):
 #   Column          Non-Null Count  Dtype  
---  ------          --------------  -----  
 0   place_name      5726 non-null   object 
 1   entity_type     5726 non-null   object 
 2   filename        5726 non-null   object 
 3   mappify_suburb  5726 non-null   object 
 4   state           5726 non-null   object 
 5   mappify_lat     5726 non-null   float64
 6   mappify_long    5726 non-null   float64
 7   partial_ratio   5726 non-null   int64  
 8   article_id      5726 non-null   int64  
dtypes: float64(2), int64(2), object(5)
memory usage: 402.7+ KB


In [58]:
#load in publication data
#need to append it to 
publication_data = pd.read_csv('/Users/fiannualamorgan/Documents/GitHub/Historical_Fires_Near_Me/Sections/Bushfire_News_Articles_19th_Century/Cleaned_Datasets/Publication_Place_Spreadsheet.csv')

In [53]:
publication_data.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5102 entries, 0 to 5101
Data columns (total 17 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   article_id       5102 non-null   int64  
 1   Lat              5025 non-null   float64
 2   Long             5025 non-null   float64
 3   Place            5038 non-null   object 
 4   newspaper_title  5102 non-null   object 
 5   State            1128 non-null   object 
 6   title            5101 non-null   object 
 7   newspaper_id     5102 non-null   int64  
 8   page             5102 non-null   object 
 9   date             5102 non-null   object 
 10  category         5102 non-null   object 
 11  words            5102 non-null   int64  
 12  illustrated      5102 non-null   object 
 13  corrections      5102 non-null   int64  
 14  snippet          5043 non-null   object 
 15  url              5087 non-null   object 
 16  page_url         5087 non-null   object 
dtypes: float64(2),

In [60]:
import random
random.choice(merged_df.article_id.to_list())

66447861

In [61]:
len(list(set(merged_df.article_id.to_list())))

2235

In [62]:
len(list(set(publication_data.article_id.to_list())))

5102

In [ ]:
import random

def get_newspaper_title(article_id_to_match, df):
    for i,row in df.iterrows():
        if str(row.article_id) == str(article_id_to_match):
            return row.newspaper_title
        else:
            continue

#print(get_newspaper_title(random.choice(merged_df.article_id.to_list()), publication_data))

#matching_lines = df[df['article_id'] == str(article_id_to_match)]
    #if matching_lines.shape[0] == 0:
        #print(matching_lines.shape)
        #return None
    #else:
        #newspaper_title_match = matching_lines.newspaper_title.to_list()[0]
        #return newspaper_title_match"""



In [ ]:
newspaper_titles = []
for i,article_id in enumerate.merged_df.article_id.to_list():
    newspaper_titles.append(get_newspaper_title(article_id, publication_data))
    print( index article id found)

In [ ]:
merged_df['newspaper_title'] = [get_newspaper_title(article_id, publication_data) for article_id in merged_df.article_id.to_list()]

In [ ]:
print(merged_df.newspaper_title)

In [99]:
for index,fruit in enumerate(['apple', 'pear', 'orange']):
    



0
1
2


In [86]:
merged_df.head()

article_id placename                          entity_type state  \
0    18801230  LOCATION     18801230-10-8991754 (1 of 1).txt   NSW   
1    18840314  LOCATION  18840314-436-254369866 (1 of 1).txt   VIC   
2    18630202  LOCATION     18630202-13-6483150 (1 of 1).txt    SA   
3    18630202  LOCATION     18630202-13-6483150 (1 of 1).txt   VIC   
4    18950225  LOCATION     18950225-30-4550862 (2 of 2).txt   NSW   

   mappify_lat  mappify_long  partial_ratio newspaper_title  
0   -31.527505    145.856320            100            None  
1   -36.972401    147.702019            100            None  
2   -35.016432    138.523661            100            None  
3   -37.813513    144.969082            100            None  
4   -35.169799    149.332208            100            None

In [70]:
results = []
for i in range(10):
    results.append(i)
print(results)

[0, 1, 2, 3, 4, 5, 6, 7, 8, 9]
